packages/teams-js/src/private/associatedApps.ts

```diff
@@ -0,0 +1,237 @@
+        throw new Error(ErrorCode.NOT_SUPPORTED_ON_PLATFORM.toString());
+      }
+
+      validateThreadId(threadId);
+
+      return sendMessageToParentAsync<[boolean, TabInstance | ErrorResponse]>(
+        'apiVersionTag', // TODO: see uses of getApiVersionTag in other files to do this correctly
+        'associatedApps.tab.addAndConfigureApp',
+        [threadId, appTypes],
+      ).then(([wasSuccessful, response]: [boolean, TabInstance | ErrorResponse]) => {
+        if (!wasSuccessful) {
+          // TODO: Can handle error codes differently here, for example if you don't want "user cancelled" to throw
+          const error = response as ErrorResponse;
+          throw new Error(`Error code: ${error.errorCode}, message: ${error.message ?? 'None'}`);
+        }
+        return response as TabInstance;
+      });
+    }
+
+    /**
+     * @hidden
+     * @internal
+     * @beta
+     * Limited to Microsoft-internal use
+     *
+     * TODO: Add full description of what this function does, ie "Allows the user to go through the tab config process again for the specified app. If
+     * no config process exists, X happens, etc."
+     *
+     * @param tab fill in details
+     * @param threadId Info about where this comes from, links to external docs if available, etc.
+     *
+     * @returns The TabInstance of the newly configured app
+     *
+     * @throws TODO: Description of errors that can be thrown from this function
+     */
+    export function reconfigure(tab: TabInstance, threadId: string): Promise<TabInstance> {
+      ensureInitialized(runtime); // TODO: add frameContext checks if this is limited to certain contexts such as content
+
+      if (!isSupported()) {
+        throw new Error(ErrorCode.NOT_SUPPORTED_ON_PLATFORM.toString());
+      }
+
+      validateTab(tab);
+      validateThreadId(threadId);
+
+      return sendMessageToParentAsync<[boolean, TabInstance | ErrorResponse]>(
+        'apiVersionTag', // TODO: see uses of getApiVersionTag in other files to do this correctly
+        'associatedApps.tab.reconfigure',
+        [tab, threadId],
+      ).then(([wasSuccessful, response]: [boolean, TabInstance | ErrorResponse]) => {
+        if (!wasSuccessful) {
+          // TODO: Can handle error codes differently here, for example if you don't want "user cancelled" to throw
+          const error = response as ErrorResponse;
+          throw new Error(`Error code: ${error.errorCode}, message: ${error.message ?? 'None'}`);
+        }
+        return response as TabInstance;
+      });
+    }
+
+    /**
+     * @hidden
+     * @internal
+     * @beta
+     * Limited to Microsoft-internal use
+     *
+     * TODO: Add full description of what this function does, ie "Renames the tab associated with an app"
+     *
+     * @param tab fill in details
+     * @param threadId Info about where this comes from, links to external docs if available, etc.
+     *
+     * @returns The TabInstance of the newly renamed app
```

Comment:
Is this renaming a tab or an app? Maybe change this to "The TabInstance of the newly renamed tab" to keep it consistent with the TODO on line 131? (I do recognize that this is largely placeholder text.)

Reply:
If this is supposed to rename the app, then maybe this function should be renamed to `renameApp`, to be consistent with `addAndConfigureApp`?

Reply:
It's renaming an app...tab? I think? This is just placeholder, I'll make sure to ask them to verify that the info is all correct